ledger-tool/src/main.rs

```diff
@@ -2947,3 +2960,58 @@ fn main() {
     measure_total_execution_time.stop();
     info!("{}", measure_total_execution_time);
 }
+
+// Override accounts without affecting the bank hash. This can useful to test whether a modified program changes the result
+// of the verify. Create a file with the contents of an account, and name it {slot}_{pubkey} and store it in a directory.
+// For example: override/279764215_A7ZG7ByDi8DpzT9Ab7CiXhvgYTJQmaDPJkMDoPitaCQV and now run:
+// ledger-tool verify --account-override=override/
+// Now at the end of slot 279764215, account A7ZG7ByDi8DpzT9Ab7CiXhvgYTJQmaDPJkMDoPitaCQV is override with the contents
+// of the file.
+fn override_accounts(
+    account_override: Option<PathBuf>,
+    cb: Option<ProcessSlotCallback>,
+) -> Option<ProcessSlotCallback> {
+    if let Some(account_override) = account_override {
+        let slot_callback = Arc::new({
+            move |bank: &Bank| {
+                if let Some(cb) = cb.clone() {
+                    cb(bank);
+                }
+
+                info!("checking for overrides in {} for slot {}", account_override.display(), bank.slot());
+
+                let prefix = format!("{}_", bank.slot());
+
+                let entries = std::fs::read_dir(&account_override).unwrap();
+
+                for entry in entries {
+                    let entry = entry.unwrap();
+
+                    if let Some(filename) = entry.path().file_name() {
+                        let filename = filename.to_string_lossy();
+
+                        if let Some(key) = filename.strip_prefix(&prefix) {
+                            let Ok(key) = Pubkey::from_str(key) else {
+                                panic!("{key} is not a valid public key")
+                            };
+
+                            let mut account = bank.get_account(&key).unwrap();
+
+                            info!("overriding {key} using {}", entry.path().display());
+
+                            let data = std::fs::read(entry.path()).unwrap();
+
+                            account.set_data_from_slice(&data);
+
+                            bank.rc.accounts.accounts_db.store_uncached(bank.slot(), &[(&key, &account)]);
```

Comment:
To you anticipate wanting/needing to reload multiple versions of the same program on the fly / at different slots ?

I'm not sure of the exact use-cases y'all have in mind, but it would seemingly be easier to just splice the program into the initial `Bank`. That is, grab the root bank shortly after this block and set it. Since it is the root, all children will inherit AND the rebuilt Bank is already frozen so no concerns about altering a `BankHash`
https://github.com/anza-xyz/agave/blob/e8ec9dc55effbf515a0f825b8165610ccc5519b7/ledger-tool/src/ledger_utils.rs#L286-L298